src/api/PluginManager.ts

```diff
@@ -16,18 +16,44 @@ export function isPluginEnabled(plugin: string) {
     return window.Aliucord.settings.get("plugins", {})[plugin] !== false;
 }
 
-export function enablePlugin(plugin: string) {
-    const plugins = window.Aliucord.settings.get("plugins", {});
-    plugins[plugin] = true;
-    window.Aliucord.settings.set("plugins", plugins);
+export async function enablePlugin(plugin: string) {
+    const settingsPlugins = window.Aliucord.settings.get("plugins", {});
+    if (isPluginEnabled(plugin)) throw new Error(`Plugin ${plugin} is already enabled.`);
+
+    settingsPlugins[plugin] = true;
+    window.Aliucord.settings.set("plugins", settingsPlugins);
+
+    const bundleZip = new ZipFile(PLUGINS_DIRECTORY + `${plugin}.zip`, 0, "r");
+    const pluginBuffer = loadPluginBundle(bundleZip);
+
+    bundleZip.close();
+
+    const pluginClass = AliuHermes.run(plugin, pluginBuffer) as typeof Plugin;
+    const enabledPlugin = plugins[plugin] = new pluginClass(plugins[plugin].manifest);
+
+    try {
+        await enabledPlugin.start();
+        enabledPlugin.enabled = true;
+    } catch (err) {
+        logger.error(`Failed while trying to start plugin: ${enabledPlugin.manifest.name}`, err);
+    }
+
     Toasts.open({ content: `Enabled ${plugin}`, source: getAssetId("Check") });
     logger.info(`Enabled plugin: ${plugin}`);
 }
 
-export function disablePlugin(plugin: string) {
-    const plugins = window.Aliucord.settings.get("plugins", {});
-    plugins[plugin] = false;
-    window.Aliucord.settings.set("plugins", plugins);
+export async function disablePlugin(plugin: string) {
+    const settingsPlugins = window.Aliucord.settings.get("plugins", {});
+
+    try {
+        await plugins[plugin].stop();
+        plugins[plugin].enabled = false;
+        settingsPlugins[plugin] = false;
+    } catch (err) {
+        logger.error(`Failed while stopping plugin: ${plugin}\n`, err);
+    }
+
+    window.Aliucord.settings.set("plugins", settingsPlugins);
```

Comment:
Does .set save to disk or there no point in setting it again?

Reply:
It saves to disk.